proto/p4/config/p4info.proto

```diff
@@ -20,6 +20,38 @@ import "google/protobuf/any.proto";
 
 package p4.config;
 
+message Documentation {
+  // A brief description of something, e.g. one sentence
+  string brief = 1;
+  // A more verbose description of something. Multiline is accepted. Markup format (if any) is TBD.
+  string description = 2;
+}
+
+// Top-level package documentation describing the forwarding pipeline config
+// Can be used to manage multiple P4 packages.
+message PkgInfo {
+    // a definitive name for this configuration, e.g. switch.p4_v1.0
+    string name = 1;
+    // configuration version, free-format string
+    string version = 2;
+    // brief and detailed descriptions
+    Documentation doc = 3;
+    // Miscellaneous metadata, free-form; a way to extend PkgInfo
+    repeated string annotations = 4;
+    // date the configuration was built or compiled
+    string build_date = 5;
+    // the toolchain used to build it, e.g. "p4c-bm2-ss v 0.0.5"
+    string compiler = 6;
```

Comment:
I had a discussion offline with other folks and one concern that came up was that fields "build_date" and "compiler" may cause the P4Info to mutate when compiling the exact same P4 program at different times or with a different p4c binary (e.g. bmv2 and some other vendor).

Right now because of how the compiler generates the P4Info, a common workflow is:
1. run each compiler backend and generate a P4Info in each case
2. **make sure all P4Info are the same** (as a sanity check) and pick one
3. send appropriate compiler output + chosen P4Info to each switch

It seems that "build_date" and "compiler" break this workflow, without being very valuable pieces of information. In particular, knowing which p4c binary (in your example "p4c-bm2-ss") should be irrelevant.

I believe that the immutability of the P4Info message as a whole is a nice property and we should try to maintain it. It seems that the most interesting pieces of information (which P4 program, which version of the P4 program, which version of the P4Info format) preserve this property.

Reply:
@teverman @wmohsin @aghaffarkhah I think this describes your concerns

Reply:
I second the comments by @antoninbas.  I also agree with an architecture string (such as "psa") vs. a target string.

Reply:
Thanks for the comments. No argument on replacing "target" with "architecture." Good suggestion.

Regarding the compiler and build_date usefulness: I was thinking that these two fields could be useful for dealing with potential inconsistencies between a device and how/when the program was compiled. Imagine some years down the road and there is a body of P4 programs which target a device, but only some of them still work due to SW (O/S, application) changes on the device. The build date and compiler might provide useful filtering criteria versus program name and version. (To me, build date is always of interest.)

To address the multi-build inconsistency, we could add a compiler flag to populate them conditionally at compile-time; they could initially be suppressed and added later as a post-processing step done to P4Info using a suitable utility.

Nevertheless I'm happy to withdraw those fields if they are objectionable. They can be added back later or conveyed as annotations until and if they become worthy.

Reply:
Done - I deleted build_date and compiler; changed target to arch.